### PR TITLE
Merged #68 : Added click's autocompletion support

### PR DESCRIPTION
**Pull request by GhostOps77**
Added the click's autocompletion support for click v7 and v8

Also added some test cases for it
And I've also segregated the test cases into separate file
I've moved all the main code to utils.py, and imported only the essential parts from the code explicitly, that the user's gonna use

Solves: #69 

And It also merges all my final changes in my main click-repl repository

After merging this PR, you can close these issues/PR

- #68
- #69 

- #64
- #65

- #74
- #80 

- #82 
- #54 
- #66 

And after all that, don't forget to update the version number

### CODE REVIEW

**Review by auvipy — COMMENTED** [review]
can we check if we can consider this package for test?
https://pypi.org/project/pytest-click/

**Comment by GhostOps77** [comment]
> can we check if we can consider this package for test? https://pypi.org/project/pytest-click/

You can check it whether it can be used for tests or not

But it will still test only the click commands, but not the click_repl package as far as I've seen from their docs

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@ click-repl
 ===
 
 [![build](https://travis-ci.org/click-contrib/click-repl.svg?branch=master)](https://travis-ci.org/click-contrib/click-repl)
-[![Tests](https://github.com/GhostOps77/click-repl/actions/workflows/tests.yml/badge.svg?branch=GhostOps77-patch-1)](https://github.com/GhostOps77/click-repl/actions/runs/4074019555/jobs/7018648133)
```
Comment (auvipy) [inline README.md:5]: why referencing your repo instead of main repo?

**Review by auvipy — CHANGES_REQUESTED** [review]

**Comment by GhostOps77** [comment]
@auvipy  Changed it, check it out

And also, I don't know what to do with the build badge, cuz there's no purpose in having it as we've remove the travis ci configs

But it's up to u now

**Comment by GhostOps77** [comment]
And I think this PR can be closed as well

#82

**Review by auvipy — APPROVED** [review]

**Comment by GhostOps77** [comment]
@auvipy hey, are u able to do the test using pytest-click?

**Pull request merged**